### PR TITLE
Uncrustified top level vital

### PR DESCRIPTION
**Pull request by quantumech3**

### CODE REVIEW

File: vital/iterator.h [hunk 1]
```diff
@@ -171,12 +180,14 @@ class base_iterator
    * Swaps function reference, current value pointer and past-end state
    * boolean.
    */
-  friend void swap( base_iterator<T, Tb> const & lhs,
-                    base_iterator<T, Tb> const & rhs )
+  friend void
+
```
Comment (mwoehlke-kitware) [inline vital/iterator.h:184]: ...must be a bug?

File: vital/iterator.h [hunk 2]
```diff
@@ -200,8 +212,8 @@ class base_iterator
    */
   base_iterator()
     : m_next_value_func( nullptr )
-    , m_cur_val_ptr( nullptr )
-    , m_at_end( true )
+      , m_cur_val_ptr( nullptr )
+      , m_at_end( true )
```
Comment (mwoehlke-kitware) [inline vital/iterator.h:216]: This is probably wrong... I suspect either a misconfiguration or unimplemented feature.

File: vital/iterator.h [hunk 3]
```diff
@@ -424,44 +439,52 @@ class const_iterator
   /**
    * Friend operator== overload for testing [non-const, const] equality.
    */
-  friend bool operator==( iterator<T> const & it,
-                          const_iterator<T> const & cit )
+  friend bool
+  operator==( iterator< T > const& it,
+              const_iterator< T > const& cit )
```
Comment (mwoehlke-kitware) [inline vital/iterator.h:444]:
```suggestion
  operator==( iterator< T > const& it, const_iterator< T > const& cit )
```

File: vital/iterator.h [hunk 4]
```diff
@@ -424,44 +439,52 @@ class const_iterator
...
   }
+
   /**
    * Friend operator== overload for testing [const, non-const] equality.
    */
-  friend bool operator==( const_iterator<T> const & cit,
-                          iterator<T> const & it )
+  friend bool
+  operator==( const_iterator< T > const& cit,
+              iterator< T > const& it )
```
Comment (mwoehlke-kitware) [inline vital/iterator.h:454]:
```suggestion
  operator==( const_iterator< T > const& cit, iterator< T > const& it )
```

File: vital/iterator.h [hunk 5]
```diff
@@ -424,44 +439,52 @@ class const_iterator
...
   }
+
   /**
    * Friend operator!= overload for testing [non-const, const] non-equality.
    */
-  friend bool operator!=( iterator<T> const & it,
-                          const_iterator<T> const & cit )
+  friend bool
+  operator!=( iterator< T > const& it,
+              const_iterator< T > const& cit )
```
Comment (mwoehlke-kitware) [inline vital/iterator.h:464]:
```suggestion
  operator!=( iterator< T > const& it, const_iterator< T > const& cit )
```

File: vital/iterator.h [hunk 6]
```diff
@@ -424,44 +439,52 @@ class const_iterator
...
   }
+
   /**
    * Friend operator!= overload for testing [const, non-const] non-equality.
    */
-  friend bool operator!=( const_iterator<T> const & cit,
-                          iterator<T> const & it )
+  friend bool
+  operator!=( const_iterator< T > const& cit,
+              iterator< T > const& it )
```
Comment (mwoehlke-kitware) [inline vital/iterator.h:474]:
```suggestion
  operator!=( const_iterator< T > const& cit, iterator< T > const& it )
```

File: vital/iterator.h [hunk 7]
```diff
@@ -531,9 +561,11 @@ class iterable
...
    */
   virtual typename const_iterator::next_value_func_t
-    get_const_iter_next_func() const = 0;
+  get_const_iter_next_func() const = 0;
 };
 
-} } // end namespaces
+} // namespace vital
+
+}   // end namespaces
```
Comment (mwoehlke-kitware) [inline vital/iterator.h:569]:
```suggestion
} // namespace kwiver
```

File: vital/math_constants.h [hunk 8]
```diff
@@ -13,17 +13,20 @@
...
+constexpr double deg_to_rad = pi / 180.0;
 
-constexpr double deg_to_rad = pi/180.0;
+constexpr double rad_to_deg = 180.0 / pi;
 
-constexpr double rad_to_deg = 180.0/pi;
+} // namespace vital
 
-} } // end namespace
+}   // end namespace
```
Comment (mwoehlke-kitware) [inline vital/math_constants.h:30]:
```suggestion
} // namespace kwiver
```

File: vital/optional.h [hunk 9]
```diff
@@ -9,991 +9,1275 @@
...
+#include <functional>
+#include <initializer_list>
+#include <stdexcept>
+#include <string>
+#include <type_traits>
+#include <utility>
+
+#define TR2_OPTIONAL_REQUIRES( ... ) typename std::enable_if< __VA_ARGS__::value, \
+                                                              bool >::type = \
+  false
```
Comment (mwoehlke-kitware) [inline vital/optional.h:25]:
```suggestion
#define TR2_OPTIONAL_REQUIRES( ... ) \
  typename std::enable_if< __VA_ARGS__::value, bool >::type = false
```

File: vital/optional.h [hunk 10]
```diff
@@ -9,991 +9,1275 @@
...
+#endif
 
-# // In C++11 constexpr implies const, so we need to make non-const members also non-constexpr
-# if (defined __cplusplus) && (__cplusplus == 201103L)
-#   define OPTIONAL_MUTABLE_CONSTEXPR
-# else
-#   define OPTIONAL_MUTABLE_CONSTEXPR constexpr
-# endif
+# \
+  // In C++11 constexpr implies const, so we need to make non-const members also non-constexpr
```
Comment (mwoehlke-kitware) [inline vital/optional.h:101]:
```suggestion
// In C++11 constexpr implies const, so we need to make non-const members also non-constexpr
```
(I have no idea what was with the original `# //`.)

File: vital/optional.h [hunk 11]
```diff
@@ -9,991 +9,1275 @@
...
+  struct has_nothrow_move_assign< X, true >
+  {
+    constexpr static bool value =
+      noexcept( std::declval< X& >() = std::declval< X && >() );
   };
 
-  constexpr static bool value = has_nothrow_move_assign<T, is_assignable<T&, T&&>::value>::value;
+  constexpr static bool value = has_nothrow_move_assign< T, is_assignable< T&,
+                                                                           T&& >::value >
+                                ::value;
```
Comment (mwoehlke-kitware) [inline vital/optional.h:179]:
```suggestion
  constexpr static bool value =
    has_nothrow_move_assign< T, is_assignable< T&, T&& >::value >::value;
```

File: vital/optional.h [hunk 12]
```diff
@@ -9,991 +9,1275 @@
...
+# define TR2_OPTIONAL_ASSERTED_EXPRESSION( CHECK, \
+                                           EXPR ) ( ( CHECK ) ? ( EXPR ) : ( [] \
+                                                                             { \
+                                                                               assert( \
+                                                                                 ! \
+                                                                                 # \
+                                                                                 CHECK ); \
+                                                                             }(), \
+                                                                             ( \
+                                                                               EXPR ) ) )
```
Comment (mwoehlke-kitware) [inline vital/optional.h:224]:
```suggestion
# define TR2_OPTIONAL_ASSERTED_EXPRESSION( CHECK, EXPR ) \
  ( ( CHECK ) ? ( EXPR ) : ( []{ assert( ! #CHECK ); }(), ( EXPR ) ) )
```

File: vital/optional.h [hunk 13]
```diff
@@ -9,991 +9,1275 @@
...
-    constexpr constexpr_storage_t( trivial_init_t ) noexcept : dummy_() {};
+  constexpr constexpr_storage_t( trivial_init_t ) noexcept : dummy_()
+  {
+  };
 
-    template <class... Args>
-    constexpr constexpr_storage_t( Args&&... args ) : value_(constexpr_forward<Args>(args)...) {}
+  template < class... Args > constexpr constexpr_storage_t( Args &&
+                                                            ... args ) : value_( constexpr_forward< Args >(
+                                                                                   args )... )
```
Comment (mwoehlke-kitware) [inline vital/optional.h:335]:
```suggestion
  template < class... Args > constexpr constexpr_storage_t( Args && ... args )
    : value_( constexpr_forward< Args >( args )... )
```

File: vital/optional.h [hunk 14]
```diff
@@ -9,991 +9,1275 @@
...
 struct constexpr_optional_base
 {
-    bool init_;
-    constexpr_storage_t<T> storage_;
+  bool init_;
+  constexpr_storage_t< T > storage_;
 
-    constexpr constexpr_optional_base() noexcept : init_(false), storage_(trivial_init) {};
+  constexpr constexpr_optional_base() noexcept : init_( false ), storage_(
+                                                   trivial_init ) {}
```
Comment (mwoehlke-kitware) [inline vital/optional.h:371]:
```suggestion
  constexpr constexpr_optional_base() noexcept 
    : init_( false ), storage_( trivial_init ) {}
```

File: vital/optional.h [hunk 15]
```diff
@@ -9,991 +9,1275 @@
...
 
-    explicit constexpr constexpr_optional_base(const T& v) : init_(true), storage_(v) {}
+  explicit constexpr constexpr_optional_base( const T& v ) : init_( true ),
+                                                             storage_( v ) {}
```
Comment (mwoehlke-kitware) [inline vital/optional.h:374]:
```suggestion
  explicit constexpr constexpr_optional_base( const T& v )
    : init_( true ), storage_( v ) {}
```

File: vital/optional.h [hunk 16]
```diff
@@ -9,991 +9,1275 @@
...
 
-    explicit constexpr constexpr_optional_base(T&& v) : init_(true), storage_(constexpr_move(v)) {}
+  explicit constexpr constexpr_optional_base( T&& v ) : init_( true ),
+                                                        storage_( constexpr_move(
+                                                                    v ) ) {}
```
Comment (mwoehlke-kitware) [inline vital/optional.h:378]:
```suggestion
  explicit constexpr constexpr_optional_base( T&& v )
    : init_( true ), storage_( constexpr_move( v ) ) {}
```

File: vital/optional.h [hunk 17]
```diff
@@ -9,991 +9,1275 @@
...
 
-    ~constexpr_optional_base() = default;
+  ~constexpr_optional_base() = default;
 };
 
-template <class T>
+template < class T >
```
Comment (mwoehlke-kitware) [inline vital/optional.h:383]: Bah, I give up on this file; the whole thing should be going away (C++17) anyway...

File: vital/vital_config.h.in [hunk 18]
```diff
@@ -10,34 +10,35 @@
  */
 
 ///////////////////////////////////////////////////////////////////////////////////////////////
-////// Do not edit this file. It is generated by CMake. All local changes will be lost. ///////
+////// Do not edit this file. It is generated by CMake. All local changes will
+/// be lost. ///////
```
Comment (mwoehlke-kitware) [inline vital/vital_config.h.in:14]: This should be massaged...

File: vital/vital_config.h.in [hunk 19]
```diff
@@ -10,34 +10,35 @@
...
+# define VITAL_SYSTEM_WIN_API ( 1 )
 
-#elif defined(__GNUC__)
+#elif defined( __GNUC__ )
 
-# define VITAL_MUST_USE_RESULT __attribute__((__warn_unused_result__))
-# define VITAL_UNUSED __attribute__((__unused__))
-# define VITAL_SYSTEM_WIN_API (0)
+# define VITAL_MUST_USE_RESULT __attribute__( ( __warn_unused_result__ ) )
+# define VITAL_UNUSED __attribute__( ( __unused__ ) )
```
Comment (mwoehlke-kitware) [inline vital/vital_config.h.in:32]: Uh... this seems dodgy, but apparently doesn't actually make the compilers happy, so I guess we can ignore it for now.

File: vital/vital_types.h [hunk 20]
```diff
@@ -62,8 +63,10 @@ typedef int64_t activity_id_t;
 typedef std::string activity_label_t;
 
 /// Global activity used to denote an undefined activity label
-const activity_label_t UNDEFINED_ACTIVITY("UNDEFINED_ACTIVITY");
+const activity_label_t UNDEFINED_ACTIVITY( "UNDEFINED_ACTIVITY" );
+
+} // namespace vital
 
-} } // end namespace
+}   // end namespace
```
Comment (mwoehlke-kitware) [inline vital/vital_types.h:70]:
```suggestion
} // namespace kwiver
```

File: vital/vital_config.h.in (hunk 18 above)
Comment (quantumech3) [reply vital/vital_config.h.in:14]: What do you mean?

**Comment by dstoup** [comment]
Jenkins test this please

**Pull request closed without merging**